homeassistant/config_entries.py

```diff
@@ -941,17 +941,26 @@ async def async_set_unique_id(
                 if progress["context"].get("unique_id") == DEFAULT_DISCOVERY_UNIQUE_ID:
                     self.hass.config_entries.flow.async_abort(progress["flow_id"])
 
-        for entry in self._async_current_entries():
+        for entry in self._async_current_entries(include_ignore=True):
             if entry.unique_id == unique_id:
                 return entry
 
         return None
 
     @callback
-    def _async_current_entries(self) -> List[ConfigEntry]:
-        """Return current entries."""
+    def _async_current_entries(self, include_ignore: bool = False) -> List[ConfigEntry]:
+        """Return current entries.
+
+        If the flow is user initiated, filter out ignored entries unless include_ignore is True.
+        """
         assert self.hass is not None
-        return self.hass.config_entries.async_entries(self.handler)
+        return [
+            entry
+            for entry in self.hass.config_entries.async_entries(self.handler)
+            if include_ignore
+            or self.source != SOURCE_USER
+            or entry.source != SOURCE_IGNORE
+        ]
```

Comment:
```suggestion
        config_entries = self.hass.config_entries.async_entries(self.handler)

        if include_ignore:
            return config_entries

        return [
            entry for entry in config_entries
            if entry.source not in (SOURCE_USER, SOURCE_IGNORE)
        ]
```

Reply:
That change sure does improve readability 😁